### PR TITLE
Rest controller

### PR DESCRIPTION
**Pull request by tiagomiotto**
- Added support for the camera and proximity sensors
- Found out a problem with the steering motor, noted in the README
- Added a motor component to poll the data from the SBrick from time to time
- Added some unit tests for the components
- Added a controller using REST requests
- Updated the README to reflect the changes

**Note:** The camera component sends the address of the video stream to the backend. Right now it is a private IP address, which means that to access it the client needs to be in the same network as the RPi. The better solution, when we have access to a modem exclusively for the RPi, would be to change that to broadcast the public IP address, since to access the stream via the public address we need to open the necssary ports, which is highly unlikely we would be able to do otherwise.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,8 +10,10 @@ For this example we are using a remote control LEGO Truck (Mercedes-Benz Arocs)
 ## Required hardware
```
Comment (necosta) [inline README.md:11]: It's not in this pull request but ... Streamming -> Streaming

Comment (tiagomiotto) [reply README.md:11]: I will check for typos again, thanks for the heads up

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+    "python.pythonPath": "/usr/bin/python"
+}
```
Comment (csazevedo) [inline .vscode/settings.json]: Shouldn't this file be removed completely?

File: README.md [hunk 3]
```diff
@@ -1,68 +1,73 @@
 # Introduction
 This repo contains a framework to extract data acquired from sensors connected to a Rpi, using the MQTT protocol, as well as provide an easy way to access those data points using a REST API.
```
Comment (csazevedo) [inline README.md]: Not this pull request but consistency on Rpi -> RPI or RPi

File: cameraClass.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+    def gen_payload_message(self,timestamp):
+        return {
+                'stream_ip': self.get_ip_address(),
+                'stream_port': "8000",
+                'timestamp': timestamp
+            }
+
+    def get_ip_address(self):
+        ip_address = ''
```
Comment (csazevedo) [inline cameraClass.py]: This line is not necessary

File: cameraClass.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+from subprocess import call
+import socket
+import time
+import json
+import os
+
+class Camera(Component):
+
+    def setup(self):
+        self.sampInterval = 100.0
```
Comment (csazevedo) [inline cameraClass.py]: Shouldn't this sampInterval be a parameter?

File: controller/rcController.py [hunk 6]
```diff
@@ -3,6 +3,7 @@
 import sys
 import json
 sys.path.append('..')
+import mqttClient
```
Comment (csazevedo) [inline controller/rcController.py]: isn't this redundant?

File: motorClass.py [hunk 7]
```diff
@@ -0,0 +1,35 @@
+from componentClass import Component
+import time
```
Comment (csazevedo) [inline motorClass.py]: Here are several libraries that are not being used. I overlooked some of the previous pull requests but I would recommend to run pylint on all files for good formatting.

File: proximityClass.py [hunk 8]
```diff
@@ -0,0 +1,84 @@
+import RPi.GPIO as GPIO
```
Comment (csazevedo) [inline proximityClass.py:1]: there are several imports that are not used

File: tests/testCamera.py [hunk 9]
```diff
@@ -0,0 +1,21 @@
+import unittest
+import time
```
Comment (csazevedo) [inline tests/testCamera.py]: import not used

File: tests/testProximity.py [hunk 10]
```diff
@@ -0,0 +1,31 @@
+import unittest
+import time
+import sys
+sys.path.append('..')
+import proximityClass
```
Comment (csazevedo) [inline tests/testProximity.py]: import not used

**Review by csazevedo — CHANGES_REQUESTED** [review]
Looks very good. I would just suggest a small change that will give you some work:
- run pylint to format the code according to PEP8.

File: cameraClass.py (hunk 5 above)
Comment (tiagomiotto) [reply cameraClass.py]: Added the functionality to configure the sampling rate in the configuration file

**Review by tiagomiotto — COMMENTED** [review]
Changes made and requirements.txt added

**Review by csazevedo — APPROVED** [review]

**Pull request merged**